pkg/reboot/calculator.go

```diff
@@ -0,0 +1,101 @@
+package reboot
+
+import (
+	"context"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/go-logr/logr"
+
+	"github.com/medik8s/self-node-remediation/pkg/utils"
+	"github.com/medik8s/self-node-remediation/pkg/watchdog"
+)
+
+const (
+	MaxTimeForNoPeersResponse = 30 * time.Second
+)
+
+type SafeTimeCalculator interface {
+	GetTimeToAssumeNodeRebooted() time.Duration
+	Start(ctx context.Context) error
+}
+
+type safeTimeCalculator struct {
+	timeToAssumeNodeRebooted                                                time.Duration
+	wd                                                                      watchdog.Watchdog
+	maxErrorThreshold                                                       int
+	apiCheckInterval, apiServerTimeout, peerDialTimeout, peerRequestTimeout time.Duration
+	log                                                                     logr.Logger
+	k8sClient                                                               client.Client
+}
+
+func New(k8sClient client.Client, wd watchdog.Watchdog, maxErrorThreshold int, apiCheckInterval, apiServerTimeout, peerDialTimeout, peerRequestTimeout, timeToAssumeNodeRebooted time.Duration) SafeTimeCalculator {
```

Comment:
IMO `New` is to generic. Maybe `NewCalculater`?

Reply:
agree. Just `New` is fine if the thing which is created matches the package name, which isn't the case here. I would even go with `NewSafeTimeCalculator`